torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -449,28 +470,21 @@ struct SymbolicShapeAnalyzer {
     }
   }
 
-  c10::SymbolicShape extractOutputShape(
-      std::unordered_map<Value*, int64_t>& symbolic_shape_values) {
-    TORCH_INTERNAL_ASSERT(graph_->outputs().size() == 1);
-    auto output = graph_->outputs().at(0);
-    TORCH_INTERNAL_ASSERT(
-        output->type()->cast<ListType>() &&
-        output->type()->cast<ListType>()->getElementType()->cast<IntType>());
-    if (output->node()->kind() == prim::Constant) {
-      auto int_list = toIValue(output)->toIntVector();
+  c10::SymbolicShape extractListShape(
+      Value* list,
+      std::unordered_map<Value*, int64_t>& symbolic_shape_values,
+      const AliasDb& db) {
+    if (list->node()->kind() == prim::Constant) {
+      auto int_list = toIValue(list)->toIntVector();
       return c10::SymbolicShape(int_list);
     }
-    // TODO: would be nice if there were easy facility to look at uses and see
-    // if they are all pure instead of instanting db.
-    AliasDb db(graph_);
-    // If it is not a single list construct or constant, bail,
-    // otherwise we cannot analyze its output and it might be modified
-    if (output->node()->kind() != prim::ListConstruct ||
-        db.hasWriters(output)) {
+    if (list->node()->kind() != prim::ListConstruct || db.hasWriters(list)) {
       GRAPH_DEBUG("Could not extract shape ", getHeader(node_));
       return c10::SymbolicShape();
     }
-    Node* list_construct = output->node();
+    // If it is not a single list construct or constant, bail,
+    // otherwise we cannot analyze its output and it might be modified
```

Comment:
Nit: is "otherwise" needed here (from grammar perspective)? Also, should this comment be before the previous check?